File: deepseek/4.Patrón de Diseño Singleton/golang/singleton.go
package main

import (
	"github.com/gin-gonic/gin"
)

func main() {
	r := gin.Default()
	r.GET("/saludar", func(c *gin.Context) {
		nombre := c.DefaultQuery("nombre", "Mundo")
		c.JSON(200, gin.H{"mensaje": "Hola, " + nombre})
	})
	r.Run()
}

/*

go mod init singleton

go get github.com/gin-gonic/gin



*/

File: deepseek/4.Patrón de Diseño Singleton/python/singleton.py
from flask import Flask, request, jsonify

app = Flask(__name__)

@app.route('/saludar', methods=['GET'])
def saludar():
    nombre = request.args.get('nombre', 'Mundo')
    return jsonify({"mensaje": f"Hola, {nombre}"})

if __name__ == '__main__':
    app.run(debug=True)